connector/MongoDBAtlasODBC.pq

```diff
@@ -51,13 +47,73 @@ GetCredentialConnectionString = () as record =>
     in
         CredentialConnectionString;
 
-// SQL-1117: Implement ProcessUri Function
-ProcessUri = (uri as text) as record => 
+NullFieldCheck = (field as any, fieldName as text ) =>
+    let 
+        EmptyRecord = [],
+        Result = if field <> null then Record.AddField(EmptyRecord, fieldName, field) else EmptyRecord
+    in
+        Result;
+
+ProcessAuthSrc = (queryEntry as any, pathEntry as any) =>
     let
-        Result = []
+        DefaultAuthSrc = "admin",
+        FieldName = "AUTH_SRC",
+        EmptyRecord = [],
+        // authSource connection option takes precedence over the defaultauthdb  value in the connection string.
+        // If both are unspecified, then authSource defaults to 'admin'.
+        AuthSrc =    if queryEntry <> null then
+                        queryEntry
+                    else if pathEntry <> null and Text.Length(Text.Remove(pathEntry, "/")) > 0 then
+                        Text.Remove(pathEntry, "/")
+                    else
+                        DefaultAuthSrc,
+        Result = Record.AddField(EmptyRecord, FieldName, AuthSrc)
     in 
         Result;
 
+ProcessUri = (uri as text) as any => 
+    let
+        DefaultPort = "27017",
+        UriParts = Uri.Parts(uri),
+        BadUri = UriParts[Host] = "" or UriParts[Fragment] <> ""
+            or UriParts[UserName] <> "" or UriParts[Password] <> "",
+        // http and https schemes automatically set port values to 80 and 443 respectively.
+        // The following checks whether the user has set the port values explicitly to 80 or 443.
+        // An unknown scheme will set the port value to -1 if it is not specified
+        Port =  if  (UriParts[Port] = 80 and not Text.Contains(uri, ":80")) or
+                    (UriParts[Port] = 443 and not Text.Contains(uri, ":443")) or
+                    UriParts[Port] = -1 then
+                    DefaultPort
+                else
+                    Number.ToText(UriParts[Port]),
+        SSL = NullFieldCheck(UriParts[Query][ssl]?, "SSL"),

```

Comment:
I was not able to figure out how to get these fields (`ssl`,`tls`,`authSource`) in a case insensitive way.  For example if the user had `authsource` all lowercase, it would not be used here.  I can create a ticket for that or do we think it's something we can address with documentation?